Essentials/src/net/ess3/api/events/teleport/TeleportAcceptEvent.java

```diff
@@ -0,0 +1,59 @@
+package net.ess3.api.events.teleport;
+
+import net.ess3.api.IUser;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerTeleportEvent;
+
+/**
+ * Event that's called when a player executes /tpaccept
+ * <p>
+ * Cancelling this event will prevent the user from teleporting.
+ */
+public class TeleportAcceptEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private final IUser sender;
+    private final IUser receiver;
+    private final PlayerTeleportEvent.TeleportCause cause;
+
+    private boolean cancelled;
+
+    public TeleportAcceptEvent(IUser sender, IUser receiver, PlayerTeleportEvent.TeleportCause cause) {
+        this.sender = sender;
+        this.receiver = receiver;
+        this.cause = cause;
+    }
+
+    public IUser getSender() {
+        return sender;
+    }
+
+    public IUser getReceiver() {
+        return receiver;
+    }
+
+    public PlayerTeleportEvent.TeleportCause getCause() {
```

Comment:
I think it's not necessary to add a cause.
It's already clear it caused by a command.

Reply:
IMO we should include a request type rather than a cause - being able to differentiate between `/tpa`, `/tpaall`, `/tpahere` might be useful.

Reply:
> 
> 
> IMO we should include a request type rather than a cause - being able to differentiate between `/tpa`, `/tpaall`, `/tpahere` might be useful.

That's a great idea!
Perhaps an enum would work in this scenario? (I'm new to the codebase so I'm not fully understand how things work...)

Reply:
An enum would be perfect for this 👍